### DISCUSSION TITLE
Client Side Tech Stack

### DISCUSSION THREAD

**Discussion by troyhunt** [HaveIBeenPwned/ux-rebuild#3] [Ideas]
The website is currently all based on Bootstrap 3 and jQuery 2 and obviously things have moved on from there, what would you do today? It'll still run on a combination of ASP.NET web app and functions on the back end, possible also making use of some Cloudflare pages. But as far as HTML, JS and CSS, what's the right place to start fresh in 2024?

**Comment by nicosammito** [comment]
Frontend wise I can definitely recommend React in combination with NextJS and some fancy hook based functionality library to create custom components like React Aria from Adobe or the primitives from Radix.

From a backend perspective it depends on the db you are using. But Ruby on Rails is a good choice for SQL based DBs

**Comment by neilcaldwell** [comment]
Looking at https://jamstackthemes.dev/ it seems that for CSS tailwind is popular, once selecting tailwind, nextJS is most popular https://jamstackthemes.dev/#ssg=next&css=tailwind 

The stand out theme, with over 5,000 stars, is https://github.com/Blazity/next-enterprise 

I'd suggest it's worth implementing a dark/light mode, for readability and general happiness of the users, it seems possible via https://tailwindcss.com/docs/dark-mode

**Comment by neilcaldwell** [comment]
Running through a cloudflare demo, this seems like a good option, with a slightly older NextJS version: 

[CloudFlare pages NextJS](https://developers.cloudflare.com/pages/framework-guides/nextjs/deploy-a-nextjs-site/) 
npm create cloudflare@latest hibp -- --framework=next
npm install -D tailwindcss
npx tailwindcss init
```undefined
npm i -D daisyui@latest
```
Then add daisyUI to your `tailwind.config.js` files:
```js
module.exports = {
  //...
  plugins: [require("daisyui")],
}
```

The dark theme at [daisyui.com](https://daisyui.com/) looks pretty similar to the current site, and modern.

**Comment by troyhunt** [comment]
I've just [published a soft launch blog post](https://www.troyhunt.com/soft-launching-and-open-sourcing-the-have-i-been-pwned-rebrand/) which is relevant to this discussion. We've gone with Bootsrap 5 and Sass, and that's about it. Since this discussion, there was also a pretty active thread on X about what direction we should go in: https://x.com/troyhunt/status/1872617198101143726

**Comment by troyhunt** [comment]
As we've now well and truly aligned on Bootstrap 5 with SASS and hand-rolled JavaScript, I'm going to close this discussion off.
